src/linkplay/utils.py

```diff
@@ -67,6 +75,50 @@ async def session_call_api_ok(
         raise LinkPlayRequestException(f"Didn't receive expected OK from {endpoint}")
 
 
+async def call_tcpuart(host: str, cmd: str) -> str | None:
+    """Get the latest data from TCP UART service."""
+    LENC = format(len(cmd), '02x')
+    HED1 = '18 96 18 20 '
+    HED2 = ' 00 00 00 c1 02 00 00 00 00 00 00 00 00 00 00 '
+    CMHX = ' '.join(hex(ord(c))[2:] for c in cmd)
+    data = None
+    _LOGGER.debug("Sending to %s TCP UART command: %s", host, cmd)
+    try:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            s.settimeout(API_TIMEOUT)
+            s.connect((host, TCPPORT))
+            s.send(bytes.fromhex(HED1 + LENC + HED2 + CMHX))
+            data = str(repr(s.recv(1024))).encode().decode("unicode-escape")
+
+        pos = data.find("AXX")
+        if pos == -1:
+            pos = data.find("MCU")
+
+        data = data[pos:(len(data)-2)]
+        _LOGGER.debug(
+            "Received from %s TCP UART command result: %s", host, data)
+        try:
+            s.close()
+        except:
+            pass
+
+    except socket.error as ex:
+        _LOGGER.debug("Error sending TCP UART command: %s with %s", cmd, ex)
+        data = None
```

Comment:
Re-raise the exception as a `LinkPlayRequestException` like done in https://github.com/Velleman/python-linkplay/blob/c5c08d147cf39ce497b261696d245081212676d9/src/linkplay/utils.py#L40

Then we won't ever have to return `None`